From: Marcin Jurkowski <marcin1j@gmail.com>
Date: Thu, 31 Oct 2013 22:33:10 +0000
Subject: [PATCH] bcm63xx: Add kernel support for Sagemcom F@ST2704V2 ADSL
 router

This adds kernel support support for Sagemcom F@st 2704 wireless ADSL
router.
It's a BCM6328-based 802.11n wireless router with USB port and ADSL2+
modem equipped with 64 MiB RAM and 8 MiB flash.

Signed-off-by: Marcin Jurkowski <marcin1j@gmail.com>
---
--- a/arch/mips/bcm63xx/boards/board_bcm963xx.c
+++ b/arch/mips/bcm63xx/boards/board_bcm963xx.c
@@ -341,6 +341,43 @@ static struct board_info __initdata boar
 		},
 	},
 };
+
+static struct board_info __initdata board_FAST2704V2 = {
+	.name				= "F@ST2704V2",
+	.expected_cpu_id		= 0x6328,
+
+	.has_pci			= 1,
+	.has_ohci0			= 1,
+	.has_ehci0			= 1,
+	.has_usbd			= 1,
+
+	.has_enetsw			= 1,
+
+	.enetsw = {
+		.used_ports = {
+			[0] = {
+				.used	= 1,
+				.phy_id = 1,
+				.name	= "Port 1",
+			},
+			[1] = {
+				.used	= 1,
+				.phy_id	= 2,
+				.name	= "Port 2",
+			},
+			[2] = {
+				.used	= 1,
+				.phy_id	= 3,
+				.name	= "Port 3",
+			},
+			[3] = {
+				.used	= 1,
+				.phy_id	= 4,
+				.name	= "Port 4",
+			},
+		},
+	},
+};
 #endif /* CONFIG_BCM63XX_CPU_6328 */
 
 /*
@@ -1404,6 +1441,7 @@ static const struct board_info __initcon
 	&board_963281TAN,
 	&board_A4001N1,
 	&board_dsl_274xb_f1,
+	&board_FAST2704V2,
 #endif
 #ifdef CONFIG_BCM63XX_CPU_6338
 	&board_96338gw,
@@ -1479,6 +1517,7 @@ static struct of_device_id const bcm963x
 	{ .compatible = "comtrend,ar-5381u", .data = &board_AR5381u, },
 	{ .compatible = "comtrend,ar-5387un", .data = &board_AR5387un, },
 	{ .compatible = "d-link,dsl-274xb-f", .data = &board_dsl_274xb_f1, },
+	{ .compatible = "sagem,f@st2704v2", .data = &board_FAST2704V2, },
 #endif
 #ifdef CONFIG_BCM63XX_CPU_6338
 	{ .compatible = "brcm,bcm96338gw", .data = &board_96338gw, },
